src/layers/io/input_layer.cpp

```diff
@@ -26,8 +26,11 @@
 
 #define LBANN_INPUT_LAYER_INSTANTIATE
 #include "lbann/layers/io/input_layer.hpp"
-#include "lbann/utils/profiling.hpp"
+
 #include "lbann/callbacks/imcomm.hpp"
+#include "lbann/execution_contexts/execution_context.hpp"
```

Comment:
Why do we need both includes here?

Reply:
Strictly speaking, we don't _need_ both. However, from an IWYU perspective, we use the `execution_context` API as well as the more specific `sgd_execution_context` API. There should be no harm to both includes. When we eventually get rid of the `sgd`-specific stuff, we'll only need the `execution_context` include.